modules/core/src/main/scala/org/tesselation/infrastructure/trust/storage/TrustStorage.scala

```diff
@@ -33,6 +33,17 @@ object TrustStorage {
       def getPublicTrust: F[PublicTrust] = getTrust.map { trust =>
         PublicTrust(trust.mapFilter { _.publicTrust })
       }
+
+      def updatePeerPublicTrustInfo(peerId: PeerId, publicTrust: PublicTrust): F[Unit] = trust.update { t =>
+        t ++ Map(peerId -> t.getOrElse(peerId, TrustInfo()).copy(peerLabels = publicTrust.labels))
```

Comment:
There is a nice function `updatedWith` for `put or update` case :)
```scala
t.updatedWith(peerId) { trustInfo =>
  Some(trustInfo.getOrElse(TrustInfo()).copy(peerLabels = publicTrust.labels))
}
```

Reply:
Fixed